File: test/test_phidget.py
from Phidget22.Phidget import *
from Phidget22.Devices.VoltageRatioInput import *
import time


def onSensorChange(self, sensorValue, sensorUnit):
    print("SensorValue: " + str(sensorValue))
    print("SensorUnit: " + str(sensorUnit.symbol))
    print("Press enter to finish")
    print("----------")


def main():
    voltageRatioInput0 = VoltageRatioInput()

    voltageRatioInput0.setIsHubPortDevice(True)
    voltageRatioInput0.setHubPort(0)
    
    voltageRatioInput0.setOnSensorChangeHandler(onSensorChange)
    
    voltageRatioInput0.openWaitForAttachment(5000)
    
    voltageRatioInput0.setSensorType(VoltageRatioSensorType.SENSOR_TYPE_1120)
    
    try:
        input("Press Enter to Stop\n")
    except (Exception, KeyboardInterrupt):
        pass
    
    voltageRatioInput0.close()

if __name__ == "__main__":
    main()

File: test/Test_Phidget.java
import com.phidget22.*;
import java.util.Scanner; //Required for Text Input
import java.io.IOException;

public class Test_Phidget {

	public static void main(String[] args) throws Exception {
		VoltageRatioInput voltageRatioInput0 = new VoltageRatioInput();

		voltageRatioInput0.setIsHubPortDevice(true);
		voltageRatioInput0.setHubPort(0);

		voltageRatioInput0.addSensorChangeListener(new VoltageRatioInputSensorChangeListener() {
			public void onSensorChange(VoltageRatioInputSensorChangeEvent e) {
				System.out.println("SensorValue: " + e.getSensorValue());
				System.out.println("SensorUnit: " + e.getSensorUnit().symbol);
				System.out.println("Press enter to finish");
				System.out.println("----------");
			}
		});

		voltageRatioInput0.open(5000);

		voltageRatioInput0.setSensorType(VoltageRatioSensorType.PN_1120);

		//Wait until Enter has been pressed before exiting
		System.in.read();

		voltageRatioInput0.close();
	}
}
